### PR TITLE
[4.0] drop jQuery

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- jQuery code can be written as plain javascript
- The code was using jQuery but didn't actually make sure that jQuery was loaded. **COMMON PATTERN**

### CODE REVIEW

File: administrator/components/com_content/tmpl/articles/default.php [hunk 1]
```diff
@@ -58,19 +58,21 @@
...
+			event.stopPropagation();
 		});
-	})(jQuery);
-";
+	  });
+	});
+})();
+JS;
 
+// @todo mode the script to a file
```
Comment (brianteeman) [inline administrator/components/com_content/tmpl/articles/default.php]: i assume this @todo you have in several of these files should say "move" not "mode"

Comment (dgrammatiko) [reply administrator/components/com_content/tmpl/articles/default.php]: The infamous mac keyboard... (not really, I just can't type 😓)

File: administrator/components/com_content/tmpl/articles/default.php [hunk 2]
```diff
@@ -58,19 +58,21 @@
...
-		{
-			$('.article-status').on('click', function(e)
-			{
-				e.stopPropagation();
-			});
+$js = <<<JS
+(function() {
+	document.addEventListener('DOMContentLoaded', function() {
+	  var elements = [].slice.call(document.querySelectorAll('.article-status'));
+
```
Comment (Quy) [inline administrator/components/com_content/tmpl/articles/default.php]: Remove whitespace.

File: administrator/components/com_content/tmpl/featured/default.php [hunk 3]
```diff
@@ -49,20 +49,22 @@
...
-		{
-			$('.article-status').on('click', function(e)
-			{
-				e.stopPropagation();
-			});
+$js = <<<JS
+(function() {
+	document.addEventListener('DOMContentLoaded', function() {
+	  var elements = [].slice.call(document.querySelectorAll('.article-status'));
+
```
Comment (Quy) [inline administrator/components/com_content/tmpl/featured/default.php]: Remove whitespace.

File: components/com_content/tmpl/categories/default.php [hunk 4]
```diff
@@ -21,22 +21,36 @@
...
-			}		
-		});
+$js = <<<JS
+(function() {
+	document.addEventListener('DOMContentLoaded', function() {
+		var categories = [].slice.call(document.querySelectorAll('.categories-list'));
+
+		categories.forEach(function(category) {
+			var buttons = [].slice.call(document.querySelectorAll('.categories-list'));
+
```
Comment (Quy) [inline components/com_content/tmpl/categories/default.php]: Remove space

File: administrator/components/com_content/tmpl/articles/default_batch_footer.php [hunk 5]
```diff
@@ -12,16 +12,24 @@
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\HTML\HTMLHelper;
 
-JFactory::getDocument()->addScriptDeclaration("
-	jQuery('#exampleModal').on('hide.bs.modal', function (e) {
+$js = <<<JS
```
Comment (wilsonge) [inline administrator/components/com_content/tmpl/articles/default_batch_footer.php]: You can actually just remove all this JS. `exampleModal` isn't the name of an ID in Joomla

Comment (dgrammatiko) [reply administrator/components/com_content/tmpl/articles/default_batch_footer.php]: are you sure?

Comment (dgrammatiko) [reply administrator/components/com_content/tmpl/articles/default_batch_footer.php]: code gone

Comment (brianteeman) [reply administrator/components/com_content/tmpl/articles/default_batch_footer.php]: exampleModal is a copy/paste error from whatever boilerplate code was used

**Comment by laoneo** [comment]
Can you make @Quy happy, then we can merge it.

**Comment by dgrammatiko** [comment]
@laoneo done

**Comment by laoneo** [comment]
Thanks!

**Comment by Scrabble96** [comment]
I'm struggling to follow this issue and can't see if jQuery is being dropped from Joomla 4.0 or not. On Bootstrap's Twitter feed on 2nd January they said: "As we start thinking about v5, what should we remove from Bootstrap? Already dropping jQuery and very likely IE10/11 in our next major version."

**Comment by ReLater** [comment]
@Scrabble96 
JQuery library will be provided by the J4 package and you can load it with simple HTMLHelper code lines like before. The same with Twitter Bootstrap.

But the deep JQuery and Bootstrap dependencies like we know them from J3 are removed from J4 core code as far as possible. Freedom of choice for extension and template developpers in J4.

**Comment by Scrabble96** [comment]
@ReLater 
Thank you. I notice, though, on analysing a website using cassiopeia (alpha 7.0-dev nightly build 11/12 Jan) that this file is loading by default from /media/vendor/jquery/js/jquery.min.js?3.3.1 as well as /media/vendor/bootstrap/js/bootstrap.bundle.min.js?4.1.3 and /media/legacy/js/jquery-noconflict.min.js?5cc8741ba543bcb5a55a949fd0617415

I can't see where the instruction to include these files is in the cassiopeia template files in order to be able to exclude them; i.e. the 'freedom of choice' you mention.

**Comment by dgrammatiko** [comment]
The templates right now are forcing both jquery and bootstrap.js. This is wrong and will be fixed as soon as the remaining dependencies get removed

**Comment by ReLater** [comment]
> I can't see where the instruction to include these files

New magic in J4
```
$wa   = $this->getWebAssetManager();
$wa->enableAsset('template.cassiopeia.' ... and so on ...
```
in combination with file `joomla.asset.json`

**Comment by thednp** [comment]
Sup. On the new `administrator/components/com_fields/helpers/fields.php` line 350

replace `if (!element.val() !== "$assignedCatids") {` with `if (!element.value !== "$assignedCatids") {`

That's the only jQuery thing I've yet noticed in the PR.

EDIT: 
Also, you can speed up the backend/frontend load BIGTIME by removing `DOMContentLoaded` listeners and adding ALL those scripts in the `<footer>`.

**Pull request merged**